### ISSUE TITLE
Azure Terraform Issues

### ISSUE THREAD

**Issue by vatsalgit5118** [prancer-io/prancer-compliance-test#317]
1) [PR-AZR-0008-TRF](https://github.com/prancer-io/prancer-compliance-test/blob/master/azure/terraform/aks.rego#L131)
- Fix the variable from `aws_attribute_absence` to `azure_attribute_absence`

2) [PR-AZR-0138-TRF](https://github.com/prancer-io/prancer-compliance-test/blob/master/azure/terraform/aks.rego#L399)
- Compliance does not check for the absence of `network_profile` attribute. If an attribute missing then compliance gets passed.

![PR-AZR-0138-TRF](https://user-images.githubusercontent.com/26707250/139443996-ae99b43c-1706-42e5-b929-c41c25c37400.png)

3) [PR-AZR-0144-TRF](https://github.com/prancer-io/prancer-compliance-test/blob/master/azure/terraform/aks.rego#L503)
- Compliance does not check for the absence of `addon_profile` and `kube_dashboard` attributes. If an attribute missing then compliance gets passed.

4) [PR-AZR-0151-TRF](https://github.com/prancer-io/prancer-compliance-test/blob/master/azure/terraform/appservice.rego#L116)
- Compliance does not check for the absence of `min_tls_version` attribute.

5) [PR-AZR-0067-TRF](https://github.com/prancer-io/prancer-compliance-test/blob/master/azure/terraform/appservice.rego#L352)
- Incorrect condition check. It should check `azure_attribute_absence["app_service_cors_not_allowing_all"]` instead of `azure_attribute_absence["app_service_uses_http_two"]`.

6) [PR-AZR-0072-TRF](https://github.com/prancer-io/prancer-compliance-test/blob/master/azure/terraform/master-compliance-test.json#L902)
- Eval contains the same value as `PR-AZR-0067-TRF`

**Comment by rezoan** [comment]
Ok i wll fix and create a PR to dev branch.

**Comment by rezoan** [comment]
@vatsalgit5118 
03. PR-AZR-0144-TRF
Compliance does not check for the absence of addon_profile and kube_dashboard attributes. If an attribute missing then compliance gets passed.
Ans: This is expected. kube_dashboard get disabled by default.

**Comment by vatsalgit5118** [comment]
@rezoan, I found some bugs in the rule

7) [PR-AZR-0078-TRF](https://github.com/prancer-io/prancer-compliance-test/blob/master/azure/terraform/appservice.rego#L591)
- Rule is not written to set this variable `azure_issue["app_service_remote_debugging_disabled"]`

8) [PR-AZR-0083-TRF](https://github.com/prancer-io/prancer-compliance-test/blob/master/azure/terraform/appservice.rego#L830)
- As per the [documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_service#python_version) Available versions are 2.7 and 3.4.

9) [PR-AZR-0002-TRF](https://github.com/prancer-io/prancer-compliance-test/blob/master/azure/terraform/vm.rego#L188) and [PR-AZR-0004-TRF](https://github.com/prancer-io/prancer-compliance-test/blob/master/azure/terraform/vm.rego#L242)
- Rule gets failed even the `allow_extension_operations` is set to `false`

10) [PR-AZR-0001-RGX](https://github.com/prancer-io/prancer-compliance-test/blob/master/azure/terraform/secrets.rego#L99)
- Testcase always gets fail even if there is no azure issue. We have to skip the test case if there is no azure issue. Remove the below code from the rego rule.
```
gl_azure_secrets = false {
   not azure_issue["gl_azure_secrets"]
}

**Comment by rezoan** [comment]
@vatsalgit5118 

7. PR-AZR-0078-TRF
Rule is not written to set this variable azure_issue["app_service_remote_debugging_disabled"]

Ans: It was working before untill [this PR](https://github.com/prancer-io/prancer-compliance-test/pull/306) get merged by you. I have reverted all the error porn changes in [this PR](https://github.com/prancer-io/prancer-compliance-test/pull/319). **Also i believe you are checking master branch accidentally. please verify development branch. it has all the latest update. if you need to verify master branch, please merge latest change from development branch to master.**

08. PR-AZR-0083-TRF
As per the documentation Available versions are 2.7 and 3.4.

Ans: added the both version check.

09. PR-AZR-0002-TRF and PR-AZR-0004-TRF
Rule gets failed even the allow_extension_operations is set to false

Ans:  **i believe you are checking master branch accidentally. please verify development branch. it has all the latest update. if you need to verify master branch, please merge latest change from development branch to master.**

PR-AZR-0001-RGX
Testcase always gets fail even if there is no azure issue. We have to skip the test case if there is no azure issue. Remove the below code from the rego rule.

Ans: Whoever copied this rule from ARM section did not modified the policy title or verified this. as far as i can remember i did fixed this in ARM long ago. been fixed now in tf as well.

------------------------------------------------------------------------------------------------------------------------
**Regarding Also, there are some rules with empty remediationDescription in master-compliance-test.json.**

Ans: As this is not a blocker and all the remediation description are related to nsg. i did saw the nsg rego rules themselves need to validate again as there are some issue.
**Creating a separate issue would be helpful here** and approve the linked PR to get going forward.

**Comment by rezoan** [comment]
closing this issue as already fixed.
